packages/keyboard-shortcuts/src/store/actions.js

```diff
@@ -24,37 +34,40 @@
 /**
  * Returns an action object used to register a new keyboard shortcut.
  *
- * @param {WPShortcutConfig} config Shortcut config.
- *
- * @return {Object} action.
+ * @param {Function} get          Atom resover.
+ * @param {Function} set          Atom updater.
+ * @param {WPShortcutConfig} config  Shortcut config.
  */
-export function registerShortcut( {
-	name,
-	category,
-	description,
-	keyCombination,
-	aliases,
-} ) {
-	return {
-		type: 'REGISTER_SHORTCUT',
-		name,
-		category,
-		keyCombination,
-		aliases,
-		description,
-	};
-}
+export const registerShortcut = ( config ) => ( { get, set } ) => {
+	const shortcutNames = get( shortcutNamesAtom );
```

Comment:
Interesting, it processes multiple atoms. I like this concpet 👍 I'm curious how it scales though.

Reply:
it can itself be an atom if we really want to push the optimization forward and the action becomes just the shortcut to the final atom.

Reply:
Still, there is this question of how many nested atoms it might absorb for the most popular actions. Think of it as a gigantic switch in the Redux reducer. I guess the fact that you import those atoms explicitly might only help to structure code better.